### PR TITLE
[jsk_spot_teleop] add dock teleop by dualsense

### PR DESCRIPTION
**Pull request by tkmtnt7000**
This PR supports spot docking by dualsense.
~~R2 is assigned as the docking button, but this button is prone to being accidentally pressed (e.g. when placed on a desk.)~~
This script supports only for Strelka currently. I don't know how Belka will behave.

**TODO**

- [x] Support for multiple robot/Dock ID to make it safe for robots that do not have a dock acceptance mechanism
- [x] ~~Docking by pressing the two buttons simultaneously to avoid malfunction~~
- [x] Use axes (cross key)

### CODE REVIEW

File: jsk_spot_robot/jsk_spot_teleop/src/joystick_teleop.cpp [hunk 1]
```diff
@@ -45,11 +47,14 @@ class TeleopManager
...
         client_stair_mode_ = nh.serviceClient<std_srvs::SetBool>("/spot/stair_mode");
         client_locomotion_mode_ = nh.serviceClient<spot_msgs::SetLocomotion>("/spot/locomotion_mode");
+        client_dock_ = nh.serviceClient<spot_msgs::Dock>("/spot/dock");
+        client_undock_ = nh.serviceClient<std_srvs::Trigger>("/spot/undock");
 
         sub_joy_ = nh.subscribe<sensor_msgs::Joy, TeleopManager>("joy", 1, &TeleopManager::callbackJoy, this);
 
         req_next_stair_mode_.data = true;
         req_next_locomotion_mode_.locomotion_mode = 4;
+        req_dock_id_.dock_id = 520; // TODO support multiple dock_id
```
Comment (mqcmd196) [inline jsk_spot_robot/jsk_spot_teleop/src/joystick_teleop.cpp]: Can you set `dock_id` as rosparam?

**Review by mqcmd196 — CHANGES_REQUESTED** [review]

File: jsk_spot_robot/jsk_spot_teleop/src/joystick_teleop.cpp (hunk 1 above)
Comment (tkmtnt7000) [reply jsk_spot_robot/jsk_spot_teleop/src/joystick_teleop.cpp]: OK. I'll use rosparam.

**Comment by tkmtnt7000** [comment]
I thought it would be a good idea to put `home_dock_id.yaml` under `/var/lib/robot`, since each robot is likely to have its own unique dock ID in the near future.
Let me know if there's a better way.

**Comment by mqcmd196** [comment]
I agree

**Comment by tkmtnt7000** [comment]
I found when `enable_button` was used, service `dock` failed. This is probably caused by `enable_button` publishing 0m/s velocity.

> - [ ] Docking by pressing the two buttons simultaneously to avoid malfunction

So I'll use cross key in place of R2 button.

File: jsk_spot_robot/jsk_spot_teleop/launch/teleop.launch [hunk 2]
```diff
@@ -31,6 +31,10 @@
         command="load"
         file="$(find jsk_spot_teleop)/config/$(arg pad_type)_teleop_twist_joy.yaml"
         />
+    <rosparam
+        command="load"
+        file="/var/lib/robot/home_dock_id.yaml"
```
Comment (sktometometo) [inline jsk_spot_robot/jsk_spot_teleop/launch/teleop.launch]: Currently, Dock ID is specified at [config.yaml](https://github.com/sktometometo/jsk_robot/blob/b7fdc3a90851d6a0ba830052a5ef1922ba05de16/jsk_spot_robot/jsk_spot_startup/scripts/config.bash#L28) as $DOCKING_STATION_ID. Can you use this?

Comment (tkmtnt7000) [reply jsk_spot_robot/jsk_spot_teleop/launch/teleop.launch]: OK. I'll use it.

**Comment by tkmtnt7000** [comment]
I checked this PR worked now.
Please let me know if there is a better way to write it that I should change.

**Pull request merged**